app/validators/sections/section_validator.py

```diff
@@ -292,3 +296,117 @@ def _get_question_variant_fields_sets(variants):
                 results["number_of_answers"].add(len(results["answer_ids"]))
 
         return results
+
+    def validate_number_of_lists(self):
+        if self.has_list_summary_with_non_item_answers() and self.has_multiple_lists():
+            self.add_error(error_messages.MULTIPLE_LISTS)
+
+    def validate_number_of_list_collectors(self):
+        if (
+            self.has_list_summary_with_non_item_answers()
+            and self.has_multiple_list_collectors()
+        ):
+            self.add_error(error_messages.MULTIPLE_LIST_COLLECTORS)
+
+    def has_list_summary_with_non_item_answers(self):
+        if summary := self.schema_element.get("summary"):
+            show_non_item_answers = summary.get("show_non_item_answers")
+            if summary.get("items") and show_non_item_answers is True:
+                return True
+
+    def has_multiple_list_collectors(self):
+        list_collector_count = 0
+        for group in self.schema_element.get("groups"):
+            for block in group.get("blocks"):
+                if block["type"] in ["ListCollector"]:
+                    list_collector_count += 1
+        if list_collector_count > 1:
+            return True
+
+    def has_multiple_lists(self):
+        lists = []
+        blocks = self.questionnaire_schema.get_blocks(type="ListCollector")
+        for block in blocks:
+            if block["for_list"] not in lists:
+                lists.append(block["for_list"])
+        if len(lists) > 1:
+            return True
+
+    def validate_show_non_item_answers_when_items_key_exists(self, summary):
+        if summary.get("show_non_item_answers") and not summary.get("items"):
+            self.add_error(error_messages.ITEMS_NOT_PRESENT)
```

Comment:
This should be done in JSON. Don't need python validation for this.

Reply:
I don't think it is quite right .`dependencies` ensure that `items` must exist when `show_non_item_answers`, but it should be `items` must exist when `show_non_item_answers` is true. You need to check the value, so you need to do it via `if` `then`.

Reply:
Updated the logic for this 👍